### PR TITLE
[dashboard] Fixes the CPU percent metrics for Dashboard process.

### PR DESCRIPTION
**Pull request by rynewang**
The metrics `ray_component_cpu_percentage` for Dashboard is always 0. This is because every time we create a fresh `psutil.Process` instance. In the document it says:

    When *interval* is 0.0 or None (default) compares process times
    to system CPU times elapsed since last call, returning
    immediately (non-blocking). That means that the first time
    this is called it will return a meaningful 0.0 value.

... so we always get a 0.0 value.

This PR changes to cache the dash proc `psutil.Process` instance in the head module class so the estimation can be non-0. Note: the readings are CPU percentage *since the last call* so it's a 5 second (`METRICS_RECORD_INTERVAL_S`) average.

I tested on my laptop, the value matches output of `ps -p $PID -o %cpu` and an external psutil python script.

### CODE REVIEW

File: dashboard/metrics_utils.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import os
+import psutil
+from ray.dashboard import k8s_utils
+
+# Are we in a K8s pod?
+IN_KUBERNETES_POD = "KUBERNETES_SERVICE_HOST" in os.environ
+
+
+def get_cpu_percent(in_k8s: bool):
+    if in_k8s:
```
Comment (rkooo567) [inline dashboard/metrics_utils.py:10]: nit; 

```
if IN_KUBERNETES_POD:
    return k8_utils...
```
?

**Review by rkooo567 — APPROVED** [review]

**Review by jjyao — CHANGES_REQUESTED** [review]

File: dashboard/metrics_utils.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        return k8s_utils.cpu_percent()
+    else:
+        return psutil.cpu_percent()
```
[the remark this answers is not in the document]
Comment (rynewang) [reply dashboard/metrics_utils.py:13]: you are totally right; I was looking at the wrong place in agent. In fact we only need to cache the psutil.Process instance. fixing....

[the remark this answers is not in the document]
Comment (rynewang) [reply dashboard/metrics_utils.py]: done

File: dashboard/metrics_utils.py (hunk 1 above)
Comment (rynewang) [reply dashboard/metrics_utils.py]: removed the whole file - this code is for system wide usage, we just want proc level usage.

File: dashboard/modules/metrics/metrics_head.py [hunk 3]
```diff
@@ -96,6 +96,7 @@ def __init__(
         self._component = "dashboard"
         self._session_name = dashboard_head.session_name
         assert self._component in AVAILABLE_COMPONENT_NAMES_FOR_METRICS
+        self.dashboard_proc = psutil.Process()
```
Comment (jjyao) [inline dashboard/modules/metrics/metrics_head.py:99]: nit: self._dashboard_proc

File: dashboard/modules/metrics/metrics_head.py [hunk 4]
```diff
@@ -305,29 +306,29 @@ def _create_default_prometheus_configs(self):
...
     @dashboard_utils.async_loop_forever(METRICS_RECORD_INTERVAL_S)
     async def record_dashboard_metrics(self):
-        dashboard_proc = psutil.Process()
         self._dashboard_head.metrics.metrics_dashboard_cpu.labels(
             ip=self._ip,
             pid=self._pid,
             Component=self._component,
             SessionName=self._session_name,
-        ).set(float(dashboard_proc.cpu_percent()) * 100)
+        ).set(float(self.dashboard_proc.cpu_percent()))
```
Comment (jjyao) [inline dashboard/modules/metrics/metrics_head.py:314]: Is removing `* 100` intentional?

**Review by jjyao — APPROVED** [review]
Can you do a quick check in other places to see if this bug exists?

File: dashboard/modules/metrics/metrics_head.py (hunk 3 above)
Comment (rynewang) [reply dashboard/modules/metrics/metrics_head.py]: done

File: dashboard/modules/metrics/metrics_head.py (hunk 4 above)
Comment (rynewang) [reply dashboard/modules/metrics/metrics_head.py]: yes it's intentional. `cpu_precent` as exported from psutil has already timed 100. It was a bug. The reading in reporter_agent also did not * 100.

**Comment by rynewang** [comment]
That's the only 2 metrics currently exposed by the dash (cpu, mem) both ok now. pls merge

**Pull request merged**